seed/seed.go

```diff
@@ -134,3 +135,36 @@ func Open(seedDir, label string) (Seed, error) {
 	// highest label one?
 	return &seed16{seedDir: seedDir}, nil
 }
+
+// ReadSystemEssential retrieves in one go information about the model
+// and essential snaps of the given types for the Core 20 recovery
+// system seed specified by seedDir and label (which cannot be empty).
+func ReadSystemEssential(seedDir, label string, essentialTypes []snap.Type, tm timings.Measurer) (*asserts.Model, []*Snap, error) {
+	if label == "" {
+		return nil, nil, fmt.Errorf("system label cannot be empty")
+	}
+	seed, err := Open(seedDir, label)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	seed20, ok := seed.(EssentialMetaLoaderSeed)
+	if !ok {
+		return nil, nil, fmt.Errorf("internal error: UC20 seed must implement EssentialMetaLoaderSeed")
+	}
+
+	// load assertions into a temporary database
+	if err := seed20.LoadAssertions(nil, nil); err != nil {
+		return nil, nil, err
+	}
+
+	// Model always succeeds after LoadAssertions
```

Comment:
Nonethelss I think we should handle err here, it ensures that this doesn't fail silently in the future if anyone re-arranges something later, or Model() starts to error out under some new conditions.

Reply:
I can do this in a follow up, I still think it was fine (one those rare cases)